### PR TITLE
ODB write: Update test reference data with Channels

### PR DESCRIPTION
**Pull request by mikecooke77**
## Description

The current code on develop does not correctly write data out when the data is of size (`Location`, `Channel`).  It currently expects only arrays of size (`Location`).  This PR fixes this issue.

The contents of the three changed files can be seen in this excel spreadsheet attached:
[odc_output_with_channels_for_PR.xlsx](https://github.com/JCSDA-internal/ioda-data/files/10970175/odc_output_with_channels_for_PR.xlsx)

Note: since producing this excel sheet pge's with missing values are now in the datafile following the merging of develop.

### Issue(s) addressed

None raised specifically in this repository.

## Acceptance Criteria (Definition of Done)

All ctests passing in the associated ioda PR at:

## Dependencies

To be merged with the ufo PR at:
- [ ] waiting on JCSDA-internal/ioda/pull/922

## Impact

None outside of ioda

### CODE REVIEW

**Review by ctgh — APPROVED** [review]

**Review by srherbener — APPROVED** [review]

**Comment by mikecooke77** [comment]
@srherbener we will have to think about how to merge this when the time comes because it will conflict with the dateTime change in https://github.com/JCSDA-internal/ioda-data/pull/141.

**Comment by srherbener** [comment]
@mikecooke77 let's merge your PRs first (jcsda-internal/ioda/pull/922 and this one), then I can resolve the conflicts in #141. After merging this PR, my plan to resolve conflicts is:
- first take the develop version of the odb test reference files
- verify that the only differences running the tests with jcsda-internal/ioda/pull/931 and #141 are the date values
    - Make sure these differences show the correct date values in the test output
- update the odb test reference files

Does this sound okay? Thanks!

**Comment by mikecooke77** [comment]
> @mikecooke77 let's merge your PRs first ([JCSDA-internal/ioda/pull/922](https://github.com/JCSDA-internal/ioda/pull/922) and this one), then I can resolve the conflicts in #141. After merging this PR, my plan to resolve conflicts is:
> 
> * first take the develop version of the odb test reference files
> * verify that the only differences running the tests with [Fix wrong date generated by the ODB writer ioda#931](https://github.com/JCSDA-internal/ioda/pull/931) and [Fix wrong date generated by the ODB writer #141](https://github.com/JCSDA-internal/ioda-data/pull/141) are the date values
>   
>   * Make sure these differences show the correct date values in the test output
> * update the odb test reference files
> 
> Does this sound okay? Thanks!

Yes that makes sense to me.  Thanks for taking the pain in this merge.

**Pull request merged**
